From: Wang Shilong <wangsl-fnst@cn.fujitsu.com>
Date: Wed, 27 Feb 2013 11:21:40 +0000
Patch-mainline: 3.8
Git-commit: 235fdb8ef2a313749cf3a42e70156c618263a388
References: FATE#312888
Subject: [PATCH] Btrfs: remove reduplicate check about root in the
 function btrfs_clean_quota_tree

The check work has been done just before the function btrfs_clean_quota_tree
is called, it is not necessary to check it again, remove it.

Signed-off-by: Wang Shilong <wangsl-fnst@cn.fujitsu.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/qgroup.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/fs/btrfs/qgroup.c
+++ b/fs/btrfs/qgroup.c
@@ -742,9 +742,6 @@ static int btrfs_clean_quota_tree(struct
 	int ret;
 	int nr = 0;
 
-	if (!root)
-		return -EINVAL;
-
 	path = btrfs_alloc_path();
 	if (!path)
 		return -ENOMEM;
